.github/workflows/create-azure-self-hosted-runners.yml

```diff
@@ -0,0 +1,119 @@
+  create-runners:
+    name: create-runner-${{ matrix.runner_index }}
+    needs: create-matrix
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJSON(needs.create-matrix.outputs.matrix) }}
+    outputs:
+      vm_name: ${{ steps.generate-vm-name.outputs.vm_name }}
+    steps:
+    - name: Generate VM name
+      id: generate-vm-name
+      run: |
+        VM_NAME="actions-runner-$(date +%Y%m%d%H%M%S%N)"
+        echo "Will be using $VM_NAME as the VM name"
+        echo "vm_name=$VM_NAME" >> $GITHUB_OUTPUT
+
+    # We can't use the octokit/request-action as we can't properly mask the runner token with it
+    # https://github.com/actions/runner/issues/475
+    - name: Generate Actions Runner token and registration URL
+      run: |
+        case "${{ inputs.runner_scope }}" in
+          "org-level")
+            ACTIONS_API_URL="https://api.github.com/repos/${{ env.ACTIONS_RUNNER_ORG }}/actions/runners/registration-token"
+            echo ACTIONS_RUNNER_REGISTRATION_URL="https://github.com/${{ env.ACTIONS_RUNNER_ORG }}" >> $GITHUB_ENV
+            ;;
+          "repo-level")
+            ACTIONS_API_URL="https://api.github.com/repos/${{ env.ACTIONS_RUNNER_ORG }}/${{ env.ACTIONS_RUNNER_REPO }}/actions/runners/registration-token"
+            echo ACTIONS_RUNNER_REGISTRATION_URL="https://github.com/${{ env.ACTIONS_RUNNER_ORG }}/${{ env.ACTIONS_RUNNER_REPO }}" >> $GITHUB_ENV
+            ;;
+          *)
+            echo "Unsupported runner scope: ${{ inputs.runner_scope }}"
+            exit 1
+            ;;
+        esac
+
+        ACTIONS_RUNNER_TOKEN=$(curl \
+          -X POST \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ secrets.GH_API_PAT }}"\
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          $ACTIONS_API_URL \
+          | jq --raw-output .token)
+        echo "::add-mask::$ACTIONS_RUNNER_TOKEN"
+        echo ACTIONS_RUNNER_TOKEN=$ACTIONS_RUNNER_TOKEN >> $GITHUB_ENV
+        
+    - name: Azure Login
+      uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+    
+    # Checkout the repo so that we can access the template files
+    - uses: actions/checkout@v3
+
+    - uses: azure/arm-deploy@v1
+      with:
+        resourceGroupName: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        template: ./azure-self-hosted-runners/azure-arm-template.json
+        parameters: ./azure-self-hosted-runners/azure-arm-template-example-parameters.json githubActionsRunnerRegistrationUrl="${{ env.ACTIONS_RUNNER_REGISTRATION_URL }}" githubActionsRunnerToken="${{ env.ACTIONS_RUNNER_TOKEN }}" postDeploymentPsScriptUrl="${{ env.POST_DEPLOYMENT_SCRIPT_URL }}" virtualMachineName=${{ steps.generate-vm-name.outputs.vm_name }} virtualMachineSize=Standard_D8pls_v5 publicIpAddressName1=${{ steps.generate-vm-name.outputs.vm_name }}-ip adminUsername=${{ secrets.AZURE_VM_USERNAME }} adminPassword=${{ secrets.AZURE_VM_PASSWORD }}
```

Comment:
`Standard_D8pls_v5` is an 8-core arm64 VM with 16GB of RAM. A machine with less specs might be sufficient too.

Reply:
Let's leave it for now, I'll monitor the usage. If I'm running out of money too fast, I'll just scale it down.